######### Modified common costmap #########

#Robot's Boundaries
footprint: [[x0, y0], [x1, y1], ... [xn, yn]]   #the center of the robot is assumed to be at (0.0, 0.0)
footprint_padding: 0.01
#robot_radius: ir_of_robot
max_obstacle_height:  0.6     # the arm is mounted on top of the robot

#GeneralParams
robot_base_frame: base_link
update_frequency: 5.0
publish_frequency: 2.0    #The frequency in Hz for the map to be publish display information.
transform_tolerance: 0.5
resolution: 0.05    #meters/cell

#Marking&Clearing
obstacle_range: 2.5    #robot will only update its map with information about obstacles that are within 2.5 meters 
raytrace_range: 3.0    #robot will attempt to clear out space in front of it up to 3.0 meters away given a sensor reading

obstacle_layer:
  enabled:              true

  #ObstacleCostmapPlugin
  track_unknown_space:  true    #true means each pixel has one of 3 states: lethal obstacle, free, or unknown
  footprint_clearing_enabled: true  #default: true
  combination_method:   1    #default: 1
 
  #VoxelCostmapPlugin
  z_resolution:         0.2   #The z resolution of the map in meters/cell.
  z_voxels:             10    #The number of voxels to in each vertical column, the height of the grid is z_resolution * z_voxels
  unknown_threshold:    15    #The number of unknown cells allowed in a column considered to be "known"
  mark_threshold:       0     #The maximum number of marked cells allowed in a column considered to be "free"
  publish_voxel_map: true    #Whether or not to publish the underlying voxel grid for visualization purposes

  #ObservationSources
  observation_sources: laser_scan_sensor point_cloud_sensor #defines a list of sensors that are going to be passing information to the costmap separated by spaces
  laser_scan_sensor: {sensor_frame: frame_name, data_type: LaserScan, topic: topic_name, marking: true, clearing: true}
  point_cloud_sensor: {sensor_frame: frame_name, data_type: PointCloud, topic: topic_name, marking: true, clearing: true}


inflation_layer:
  enabled:              true
  cost_scaling_factor:  5.0  #increasing the factor will decrease the resulting cost values(default: 10)
  inflation_radius:    0.55 #max. distance from an obstacle at which costs are incurred for planning paths.

static_layer:
  enabled:              true